modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientHandlerModule.java

```diff
@@ -177,6 +182,7 @@ public ClientHandlerModule(
         this.clock = clock;
         this.schemaSyncService = schemaSyncService;
         this.catalogService = catalogService;
+        this.primaryReplicaTracker = new ClientPrimaryReplicaTracker(placementDriver, igniteTables, catalogService, clock);
```

Comment:
Catalog does not provide the number of partitions for a given table, which is needed in `initReplicasForTableAsync`. That's the only reason to pull `igniteTables` in there. 

It would be great to get rid of `igniteTables` if we can retrieve the partition count from `PlacementDriver` or `CatalogService`, but I could not find a way to do that.

Reply:
`igniteTables` is only used to get `partitions` - I missed the fact that `catalogService` can provide that. Will remove.